datahub-graphql-core/src/main/resources/gms.graphql

```diff
@@ -19,6 +19,71 @@ interface Entity {
     GMS Entity Type
     """
     type: EntityType!
+
+    """
+    Edges extending from this entity.
+    """
+    relationships(types: [String!]!, direction: RelationshipDirection!, start: Int, count: Int): EntityRelationshipsResult
+}
+
+"""
+Result storing relationship information associated with a particular entity.
+"""
+type EntityRelationshipsResult {
+
+  """
+  Start offset of the result set
+  """
+  start: Int
+
+  """
+  Number of results in the returned result set
+  """
+  count: Int
+
+  """
+  Total number of results in the result set
+  """
+  total: Int
+
+  """
+  Relationship edges in the result set
+  """
+  relationships: [EntityRelationshipV2!]!
+}
+
+"""
+The second version of EntityRelationship model, where Entity can be any entity.
+TODO - Migrate all entity relationships to this more generic model
+"""
+type EntityRelationshipV2 {
```

Comment:
since V2 is a superset of V1, thoughts on just adding `type` and `direction` to V1 but leaving them optional? Otherwise we will be saddled w/ this V2 suffix forever

Reply:
It doesn't really work. EntityRelationship expects an EntityWithRelationships which messes everything up